assets/controllers/form-related-links_controller.js

```diff
@@ -0,0 +1,50 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    static targets = ['relatedContainer'];
+
+    static values = {
+        index: Number,
+        link: String,
+    };
+
+    connect() {
+        const container = this.element;
+        container
+            .querySelectorAll('.related-link-item')
+            .forEach((item) => {
+                this.#addButtonDeleteLink(item);
+            });
+    }
+
+    addRelatedElement() {
+        const nodeLink = this.#htmlToNode(this.linkValue.replace(
+            /__name__/g,
+            this.indexValue,
+        ));
+
+        this.#addButtonDeleteLink(nodeLink);
+
+        this.relatedContainerTarget.appendChild(nodeLink);
+        this.indexValue++;
+    }
+
+    #addButtonDeleteLink(item) {
+        const removeFormButton = document.createElement('button');
+        removeFormButton.innerText = '⌫';
```

Comment:
I think overwriting `innerHtml` with the font awesome trashcan icon (https://fontawesome.com/icons/trash?f=classic&s=solid) would be better. Then you could also make it a button:
```html
<div class="btn btn__secondary">
    <i class="fa-solid fa-trash"></i>
</div>
```

Reply:
Modified